packages/react/src/components/footer/footerNavigation/FooterNavigation.tsx

```diff
@@ -1,39 +1,51 @@
-import React, { useContext, useEffect } from 'react';
+import React, { Children, cloneElement } from 'react';
 
 // import core base styles
 import 'hds-core';
 import styles from './FooterNavigation.module.scss';
-import { FooterContext } from '../FooterContext';
-import { FooterNavigationVariant } from '../Footer.interface';
 import classNames from '../../../utils/classNames';
+import { getChildElementsEvenIfContainerInbetween } from '../../../utils/getChildren';
+import { FCWithName } from '../../../common/types';
+import { useMediaQueryLessThan } from '../../../hooks/useMediaQuery';
+import { FooterVariant } from '../Footer.interface';
 
 export type FooterNavigationProps = React.PropsWithChildren<{
   /**
-   * The aria-label for the `<nav>` element. Describes the navigation to screen reader users.
+   * Description of the navigation for screen readers.
    */
-  navigationAriaLabel?: string;
-  /**
-   * Defines how the navigation items will be displayed in the footer
-   *
-   * Supported values:
-   *
-   * `default` - Items will be displayed beneath the logo and title
-   *
-   * `minimal` - Items will be displayed inline with the logo and title. Intended to be used with 4 items or less.
-   *
-   * `sitemap` - Items will be displayed beneath the logo and title as groups with items and sub-items...
-   */
-  variant?: FooterNavigationVariant;
+  ariaLabel?: string;
 }>;
 
-export const FooterNavigation = ({ children, navigationAriaLabel, variant }: FooterNavigationProps) => {
-  const { dispatch } = useContext(FooterContext);
-
-  useEffect(() => dispatch({ type: 'NAVIGATION_VARIANT', value: variant }), [dispatch, variant]);
+export const FooterNavigation = ({ children, ariaLabel }: FooterNavigationProps) => {
+  const isSmallerThanLargeScreen = useMediaQueryLessThan('l');
+  const groups = getChildElementsEvenIfContainerInbetween(children).filter(
+    (child) => (child.type as FCWithName).componentName === 'FooterNavigationGroup',
+  );
+  const hasGroups = groups && groups.length > 0;
 
   return (
-    <nav className={classNames(styles.navigation, styles[variant])} aria-label={navigationAriaLabel}>
-      {children}
+    <nav
+      className={classNames(styles.navigation, hasGroups && !isSmallerThanLargeScreen && styles.sitemap)}
+      aria-label={ariaLabel}
+    >
+      {hasGroups && !isSmallerThanLargeScreen && (
+        <div className={styles.groups}>
+          {Children.map(groups, (child, index) => {
+            return cloneElement(child, {
+              key: index,
+            });
+          })}
+        </div>
+      )}
+      {hasGroups &&
+        isSmallerThanLargeScreen &&
+        Children.map(groups, (group, index) => {
+          return cloneElement(group.props.children[0], {
+            key: index,
+            variant: FooterVariant.Navigation,
+          });
+        })}
+      {!hasGroups && children}
```

Comment:
Is it possible that there would be a non-link element as a first group item?

Reply:
It could be possible that somebody wanted to place links inside a container with own styling but I don't think we'd want that? Can you think of some other cases?

Reply:
Not really. I was thinking could it be more clear when and what it renders on small screens, for example
```tsx
 isSmallerScreen ? renderOnlyMainLinks
```
etc. ?

Reply:
Gotcha 👍 That would make it clearer. I pushed an update.